TestApp/Login/Sources/LoginViews/LoginScreen.swift

```diff
@@ -0,0 +1,61 @@
+import SwiftUI
+import Combine
+import Navigation
+import Login
+import UserStorage
+
+enum Route {
+    case dummySheet
+}
+
+@MainActor public final class LoginScreenViewModel: ObservableObject {
+
+    @Published var route: Route?
+
+    let userStorage: any UserStoring
+    let dismissalPublisher = PassthroughSubject<Void, Never>()
+
+    public init(userStorage: any UserStoring) {
+        self.userStorage = userStorage
+    }
+
+    func didConfirmLogin() {
+        userStorage.isLoggedIn = true
+    }
+}
+
+public struct LoginScreen: View {
+
+    @ObservedObject var viewModel: LoginScreenViewModel
+    @Environment(\.presentationMode) var presentationMode
+
+    public init(viewModel: LoginScreenViewModel) {
+        self.viewModel = viewModel
+    }
+
+    public var body: some View {
+        NavigationView {
+            VStack(spacing: 20) {
+                Button("Finish login") {
+                    viewModel.didConfirmLogin()
+                    presentationMode.wrappedValue.dismiss()
```

Comment:
The dismissal is probably not needed - the didConfirmLogin should probably send the signal to the dismissalPublisher as a part of itself (it would also work for the `onDeeplink`)

Reply:
This one isn't. Interestingly, it doesn't work from the `onDeeplink` handler 🤔